### DISCUSSION TITLE
Where are my generated files? I can't find them or import anything from them

### DISCUSSION THREAD

**Discussion by boofhead** [ExpediaGroup/graphql-kotlin#1316] [Q&A] [answered]
I have this in my `build.gradle .kts`

```
graphql {
    client {
        endpoint = "https://domain.com/graphql"
        headers = mapOf("x-api-key" to "XXX-MY-API-KEY")
        packageName = "com.domain.graphql.generated"
        serializer = GraphQLSerializer.JACKSON
        queryFileDirectory = project.buildDir.absolutePath
    }
}
```

names changed to protect the innocent.

schema generates from introspection as expected and looks good. Client generation task runs with no errors but there is no generated code anywhere I can see and nothing I can import to use.

I know I'm being stupid here but what am I doing wrong??

**Comment by smyrick** [comment, marked as answer]
For Gradle it would be at `<project>/build/generated/...`

**Reply by boofhead** [reply]
Hi @smyrick ,

Thanks for getting back to me. That's where I expected they would be but for some reason they are not.

It won't run for me without the `queryFileDirectory` input (which is not used in the example). The only other difference I can see is that I am using `endpoint` instead of `sdlEndpoint` but as I said the `schema.graphql` is being generated and looks as I expect so I don't think that's the reason it's not working.

Let me know if I can give you more info

Robin

**Reply by boofhead** [reply]
Ok, I was being stupid, didn't realise that the queries need to be manually defined, makes sense I can see would be impossible to do directly from introspection

**Reply by dariuszkuc** [reply]
@boofhead client generation should fail if there are no query files to process (https://github.com/ExpediaGroup/graphql-kotlin/blob/master/plugins/graphql-kotlin-gradle-plugin/src/main/kotlin/com/expediagroup/graphql/plugin/gradle/tasks/AbstractGenerateClientTask.kt#L157), if that is not the case please open an issue with a link to a repo that reproduces the error
